### PR TITLE
Update processor tests and remove unused helper code

### PR DESCRIPTION
**Pull request by danesfeder**
Just a quick cleanup - I noticed some of this code was unused / not needed anymore after the native integration.

### CODE REVIEW

**Comment by danesfeder** [comment]
@devotaaabel this is ready for review when you have a chance to 👀thanks!

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  public void onRun_buildNewRouteProgressReceivesStatusAndRoute() {
+    NavigationRouteProcessor processor = mock(NavigationRouteProcessor.class);
+    NavigationStatus status = buildMockStatus();
+    DirectionsRoute route = mock(DirectionsRoute.class);
+    RouteProcessorRunnable runnable = buildRouteProcessorRunnableWith(processor, status, route);
+    runnable.updateRawLocation(mock(Location.class));
+
+    runnable.run();
+
+    verify(processor).buildNewRouteProgress(status, route);
```
Comment (devotaaabel) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java]: Maybe also verify `routeProcessor.updatePreviousRouteProgress` is called as well?

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      navigation,
+      mock(Handler.class),
+      mock(Handler.class),
+      mock(RouteProcessorBackgroundThread.Listener.class)
+    );
+  }
+
+  private RouteProcessorRunnable buildRouteProcessorRunnableWith(NavigationEngineFactory factory,
+                                                                 NavigationStatus status) {
+    MapboxNavigationOptions options  = MapboxNavigationOptions.builder().build();
```
Comment (devotaaabel) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java]: Are you opposed to using fakers to create non-mocked objects like `MapboxNavigationOptions`? Like this for example: https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/instruction/BannerComponentsFaker.java

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java (hunk 1 above)
Comment (danesfeder) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java]: @devotaaabel I added another test for this 👍

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java (hunk 2 above)
Comment (danesfeder) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java]: @devotaaabel would this be to be more explicit regarding naming within the test?

Comment (devotaaabel) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteProcessorRunnableTest.java]: @danesfeder It's mostly just to remove the clutter from the actual test classes.

**Comment by danesfeder** [comment]
@devotaaabel thanks!

**Pull request merged**